builtins/match.go

```diff
@@ -34,9 +38,65 @@ func matchRecord(a, b *runtime.Record) (bool, error) {
 			}
 		} else {
 			return false, runtime.Errorf("Value mismatch: field %s in second Record not found in first", k)
+		}
+	}
+	return true, nil
+}
 
+func matchRecordOf(val, pat []runtime.Object) (bool, error) {
+	i, back_i := 0, -1
+	j, back_j := 0, -1
+	for i < len(val) && j < len(pat) {
+		valueAtI := val[i]
+		patternAtJ := pat[j]
+		i++
+		j++
+
+		switch patternAtJ {
+		case runtime.AnyOrNone:
+			if j == len(pat) { // Optimize trailing * case
```

Comment:
Prioritize readability before optimization.

If this condition is not required for the algorithm you should get rid of.